### PR TITLE
cephadm: Add Zypper packager (openSUSE/SLES)

### PR DESCRIPTION
**Pull request by krig**
Adds a basic Packager for Zypper, enabling the `prepare-host` command on openSUSE/SLES.

# Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -2949,6 +2950,99 @@ class YumDnf(Packager):
...
+        if os.path.exists(self.repo_path()):
+            os.unlink(self.repo_path())
+
+    def install(self, ls):
+        logger.info('Installing packages %s...' % ls)
+        call_throws([self.tool, 'in', '-y'] + ls)
+
+        if 'chrony' in ls:
+            # ensure chrony service is enabled/started
+            call_throws(['systemctl', 'enable', '--now', 'chronyd'])
```
Comment (liewegas) [inline src/cephadm/cephadm]: should we put this in the caller?  we can add a chrony_service_name method to each packager.

Comment (krig) [reply src/cephadm/cephadm]: Yeah, I considered that, maybe adding a `enable_service` method to the Packager. I'll look at it.

Comment (krig) [reply src/cephadm/cephadm]: It now tries to install chrony and then enable the first time sync service it finds on the system.. maybe not perfect.

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
```
cephadm:540: error: Missing return statement
cephadm:557: error: Type signature has too many arguments
cephadm:2312: error: Argument 1 to "get_unit_name" has incompatible type "Optional[str]"; expected "str"
```

**Comment by krig** [comment]
@liewegas 
> 
> 
> ```
> cephadm:540: error: Missing return statement
> cephadm:557: error: Type signature has too many arguments
> cephadm:2312: error: Argument 1 to "get_unit_name" has incompatible type "Optional[str]"; expected "str"
> ```

Ah yes, I fixed the incorrect mypy statements but not the issues that had been hidden by them, I'll fix those as well.

**Comment by krig** [comment]
fixed the mypy errors now.

**Pull request merged**